src/Modal/index.jsx

```diff
@@ -214,54 +202,60 @@ class Modal extends React.Component {
-                    aria-labelledby={this.closeModalButtonId}
-                    onClick={this.close}
-                    inputRef={this.setFirstFocusableElement}
-                    onKeyDown={this.handleKeyDown}
-                  >
-                    <Icon className="fa fa-times js-close-modal-on-click" />
-                  </Button.Deprecated>
+            <FocusOn
+              enabled={open}
+              onClickOutside={this.close}
+              onEscapeKey={this.close}
+            >
+              <div
+                className="modal-content"
+                // if the modal doesn't contain any tabbable elements, make this element programmatically focusable.
+                {...(!hasTabbableElements ? { tabIndex: -1 } : {})}
+              >
+                <div className="modal-header">
+                  <h2 className="modal-title" id={this.headerId}>{title}</h2>
+                  {renderHeaderCloseButton && (
+                    <Button.Deprecated
+                      className="p-1"
+                      onClick={this.close}
+                    >
+                      <Icon className="fa fa-times" screenReaderText={closeText} />
+                    </Button.Deprecated>
+                  )}
+                </div>
+                <div className="modal-body" ref={this.modalBodyRef}>
+                  {this.renderBody()}
+                </div>
+                {renderModalFooter && (
+                  <div className="modal-footer">
+                    {renderDefaultCloseButton && (
+                      <Button
+                        variant="link"
+                        onClick={this.close}
+                      >
+                        {closeText}
+                      </Button>
+                    )}
+                    {this.renderButtons()}
+                  </div>
                 )}
               </div>
-              <div className="modal-body">
-                {this.renderBody()}
-              </div>
-              <div className="modal-footer">
-                {this.renderButtons()}
-                {renderDefaultCloseButton && (
-                  <Button.Deprecated
-                    id={this.closeModalButtonId}
-                    buttonType="secondary"
-                    className="js-close-modal-on-click"
-                    onClick={this.close}
-                    inputRef={this.setCloseButton}
-                    onKeyDown={this.handleKeyDown}
-                  >
-                    {this.props.closeText}
-                  </Button.Deprecated>
-                )}
-              </div>
-            </div>
+            </FocusOn>
           </div>
         </div>
-      </div>
+      </>
     );
   }
 
   render() {
-    return ReactDOM.createPortal(
-      this.renderModal(),
-      this.el,
-    );
+    if (this.el) {
+      return ReactDOM.createPortal(
+        this.renderModal(),
+        this.el,
+      );
+    }
+    return null;
```

Comment:
Just a nitpick on style: My preference is to use an early return in scenarios like these. https://medium.com/better-programming/are-early-returns-any-good-eed4b4d03866

The gist is "write functions that return the expected positive result at the end of the function." So in this case:

```
if (!this.el) {
  return null;
}

return ReactDOM.createPortal(
  this.renderModal(),
  this.el,
);

Reply:
Ah yes, I usually prefer early returns too :) I will update!